elements/pf-popover/pf-popover.ts

```diff
@@ -0,0 +1,389 @@
+ *          Popover min-width
+ *          {@default `20.75rem`}
+ * @cssprop {<number>} --pf-c-popover--c-button--Right
+ *          Close button right position
+ *          {@default `0}
+ * @cssprop {<number>} --pf-c-popover--c-button--Top
+ *          Close button top position
+ *          {@default `0`}
+ * @cssprop {<length>} --pf-c-popover--c-button--sibling--PaddingRight
+ *          Padding between close button and its immediate sibling
+ *          {@default `3rem`}
+ * @cssprop {<length>} --pf-c-popover__title-icon--MarginRight
+ *          Heading icon right margin
+ *          {@default `0.5rem`}
+ * @cssprop {<length>} --pf-c-popover__title--FontSize
+ *          Header font-size
+ *          {@default `1rem`}
+ * @cssprop {<length>} --pf-c-popover__title--MarginBottom
+ *          Header bottom margin
+ *          {@default `0.5rem`}
+ * @cssprop {<number>} --pf-c-popover__title--LineHeight
+ *          Header line height
+ *          {@default `1.5`}
+ * @cssprop {<string>} --pf-c-popover__title--FontFamily
+ *          Header font-family
+ *          {@default `'RedHatDisplay', 'Overpass', overpass, helvetica, arial, sans-serif`}
+ * @cssprop {<length>} --pf-c-popover__footer--MarginTop
+ *          Footer top margin
+ *          {@default `1rem`}
+ * @cssprop {<color>} --pf-c-popover--m-default__title-text--Color
+ *          Default alert heading color
+ *          {@default `#003737`}
+ * @cssprop {<color>} --pf-c-popover--m-default__title-icon--Color
+ *          Default alert icon color
+ *          {@default `#009596`}
+ * @cssprop {<color>} --pf-c-popover--m-info__title-text--Color
+ *          Default alert heading color
+ *          {@default `#002952`}
+ * @cssprop {<color>} --pf-c-popover--m-info__title-icon--Color
+ *          Default alert icon color
+ *          {@default `#2b9af3`}
+ * @cssprop {<color>} --pf-c-popover--m-warning__title-text--Color
+ *          Default alert heading color
+ *          {@default `#795600`}
+ * @cssprop {<color>} --pf-c-popover--m-warning__title-icon--Color
+ *          Default alert icon color
+ *          {@default `#f0ab00`}
+ * @cssprop {<color>} --pf-c-popover--m-success__title-text--Color
+ *          Default alert heading color
+ *          {@default `#1e4f18`}
+ * @cssprop {<color>} --pf-c-popover--m-success__title-icon--Color
+ *          Default alert icon color
+ *          {@default `#3e8635`}
+ * @cssprop {<color>} --pf-c-popover--m-danger__title-text--Color
+ *          Default alert heading color
+ *          {@default `#a30000`}
+ * @cssprop {<color>} --pf-c-popover--m-danger__title-icon--Color
+ *          Default alert icon color
+ *          {@default `#c9190b`}
+ */
+@customElement('pf-popover')
+export class PfPopover extends LitElement {
+  static readonly styles = [styles];
+
+  // todo: alertSeverityScreenReaderText
+  // todo: hideOnOutsideClick
+  // todo: withFocusTrap
+  @property({ reflect: true }) position: Placement = 'top';
```

Comment:
please add JSDoc blocks for all public props and methods, and separate them by an empty line